Tests/PackageLoadingTests/TargetSourcesBuilderTests.swift

```diff
@@ -99,26 +99,118 @@ class TargetSourcesBuilderTests: XCTestCase {
             "/some/path/toBeCopied/cool/hello.swift",
         ])
 
-        let diags = DiagnosticsEngine()
-
         let somethingRule = FileRuleDescription(
             rule: .processResource,
             toolsVersion: .minimumRequired,
             fileTypes: ["something"])
 
+        build(target: target, additionalFileRules: [somethingRule], toolsVersion: .v5, fs: fs) { _, _, _ in
+            // No diagnostics
+        }
+    }
+
+    func testResourceConflicts() throws {
+        // Conflict between processed resources.
+
+        do {
+            let target = TargetDescription(name: "Foo", resources: [
+                .init(rule: .process, path: "Resources")
+            ])
+
+            let fs = InMemoryFileSystem(emptyFiles:
+                "/Resources/foo.txt",
+                "/Resources/Sub/foo.txt"
+            )
+
+            build(target: target, toolsVersion: .vNext, fs: fs) { _, _, diagnostics in
+                diagnostics.check(diagnostic: "multiple resources named 'foo.txt' in target 'Foo'", behavior: .error)
+            }
+        }
+
+        // Conflict between processed and copied resources.
+
+        do {
+            let target = TargetDescription(name: "Foo", resources: [
+                .init(rule: .process, path: "Processed"),
+                .init(rule: .copy, path: "Copied/foo.txt"),
+            ])
+
+            let fs = InMemoryFileSystem(emptyFiles:
+                "/Processed/foo.txt",
+                "/Copied/foo.txt"
+            )
+
+            build(target: target, toolsVersion: .vNext, fs: fs) { _, _, diagnostics in
+                diagnostics.check(diagnostic: "multiple resources named 'foo.txt' in target 'Foo'", behavior: .error)
```

Comment:
Hmm, is it possible to improve the error here? It might be difficult for the user to figure out how the conflict is happening. Maybe we can list the file paths it picked up?

Reply:
Check out the new changes. I think you'll like it :)

Reply:
😁